### PR TITLE
Create deep-dive-blog-posts.md

### PR DESCRIPTION
**Pull request by mattwarren**
Start collating a list of blog posts the provide a 'deep-dive' into the CoreCLR source code (as per an off-line discussion with @richlander)

### CODE REVIEW

**Comment by richlander** [comment]
Thanks! Want to add a link in the Documentation/README.md file?

Sorry for the delay. The 2.0 release took me completely offline.

**Comment by mattwarren** [comment]
> Thanks! Want to add a link in the Documentation/README.md file?

Done, I put it in the BOTR section, hope that's okay?

File: Documentation/deep-dive-blog-posts.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## List of blog posts that provide a 'deep-dive' into the CoreCLR source code
+
+### Posts take a high-level look at the entire source:
```
Comment (svick) [inline Documentation/deep-dive-blog-posts.md]: Missing "that"? I.e. "Posts that take a high-level look at the entire source".

Comment (mattwarren) [inline Documentation/deep-dive-blog-posts.md]: Thanks for spotting that, it's fixed now

**Comment by karelz** [comment]
We should probably add also link to http://tooslowexception.com/zero-garbage-collector-for-net-core/
What do you think @mattwarren?

**Comment by mattwarren** [comment]
@karelz good idea, I was a bit selfish and only added my own posts! 

Here's a list of other posts that discuss the CoreCLR source (I'll send a PR to add them to the doc):

- [How does Object.GetType() really work?](http://tooslowexception.com/how-does-gettype-work/)
- [Zero Garbage Collector for .NET Core](http://tooslowexception.com/zero-garbage-collector-for-net-core/)
- [.NET Generics under the hood](https://alexandrnikitin.github.io/blog/dotnet-generics-under-the-hood/)
- [Hoisting in .NET Explained](https://alexandrnikitin.github.io/blog/hoisting-in-net-explained/)
- [Garbage collection and variable lifetime tracking](https://blogs.msdn.microsoft.com/seteplia/2017/05/09/garbage-collection-and-variable-lifetime-tracking/)
- [Managed object internals, Part 1. The layout](https://blogs.msdn.microsoft.com/seteplia/2017/05/26/managed-object-internals-part-1-layout/)
- [Embedding CoreCLR in your C/C++ application](http://yizhang82.me/hosting-coreclr)
- [Sharing .NET generic code under the hood](http://yizhang82.me/dotnet-generics-sharing)
- [The origin of GetHashCode in .NET](http://codingsight.com/the-origin-of-gethashcode-in-net/)
- [Sorting in .NET](http://codingsight.com/sorting-in-net/)
- [CoreCLR source discovery (a) what is the Object](http://www.cnblogs.com/zkweb/p/6244934.html) (in Chinese, [English translation](http://www.microsofttranslator.com/bv.aspx?from=&to=en&a=http%3A%2F%2Fwww.cnblogs.com%2Fzkweb%2Fp%2F6244934.html))
- [CoreCLR source discovery (ii) what is the new](http://www.cnblogs.com/zkweb/p/6266117.html) (in Chinese, [English translation](http://www.microsofttranslator.com/bv.aspx?from=&to=en&a=http%3A%2F%2Fwww.cnblogs.com%2Fzkweb%2Fp%2F6266117.html))
- [CoreCLR source discovery(iii) internal implementation of GC memory allocator](http://www.cnblogs.com/zkweb/p/6379080.html) (in Chinese, [English translation](http://www.microsofttranslator.com/bv.aspx?from=&to=en&a=http%3A%2F%2Fwww.cnblogs.com%2Fzkweb%2Fp%2F6379080.html))
- [CoreCLR source exploration (d) analysis of the internal implementation of GC memory collection review](http://www.cnblogs.com/zkweb/p/6625049.html) (in Chinese, [English translation](http://www.microsofttranslator.com/bv.aspx?from=&to=en&a=http%3A%2F%2Fwww.cnblogs.com%2Fzkweb%2Fp%2F6625049.html))

**Comment by karelz** [comment]
Great, thanks @mattwarren! Incidentally I just remembered this yesterday and figured my reply got lost :)
Please tag me on the PR, I'd like to ask @jkotas to skim the articles to make sure we add links only to those which are not misleading.

**Comment by jkotas** [comment]
The articles tend to be point in a time drill downs into the implementation. They do not always match the live .NET Core if the implementation changed.

Otherwise the articles in your list look fine.

**Pull request merged**